main.go

```diff
@@ -324,3 +367,278 @@ func printList() {
+		err = makeBPFObject(bpfObjInstallPath)
+		if err != nil {
+			return "", err
+		}
+		if debug {
+			fmt.Printf("successfully built ebpf obj file into: %s\n", bpfObjInstallPath)
+		}
+		bpfObjFilePath = bpfObjInstallPath
+	}
+
+	if bpfObjFilePath == "" {
+		return "", fmt.Errorf("could not find or build the bpf object file")
+	}
+	return bpfObjFilePath, nil
+}
+
+// unpackBPFBundle unpacks the bundle (tar(gzip(b64))) into the provided directory
+func unpackBPFBundle(dir string) error {
+	if bpfBundleInjected == "" {
+		return fmt.Errorf("missing embedded data")
+	}
+	b64Reader := base64.NewDecoder(base64.RawStdEncoding, strings.NewReader(bpfBundleInjected))
+	gzReader, err := gzip.NewReader(b64Reader)
+	if err != nil {
+		return err
+	}
+	tarReader := tar.NewReader(gzReader)
+	for true {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		switch header.Typeflag {
+		case tar.TypeDir:
+			//skip directories
+		case tar.TypeReg:
+			outFile, err := os.Create(filepath.Join(dir, filepath.Base(header.Name)))
+			if err != nil {
+				return err
+			}
+			defer outFile.Close()
+			if _, err := io.Copy(outFile, tarReader); err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unknown tar type: %v", header)
+		}
+	}
+	return nil
+}
+
+// makeBPFObject builds the ebpf object from source code into the provided path
+func makeBPFObject(outFile string) error {
+	// drop capabilities for the compilation process
+	cap, err := getSelfCapabilities()
+	if err != nil {
+		return err
+	}
+	capNew, err := capability.NewPid2(0)
+	if err != err {
+		return err
+	}
+	capNew.Clear(capability.BOUNDS)
+	err = capNew.Apply(capability.BOUNDS)
+	if err != err {
+		return err
+	}
+	defer cap.Apply(capability.BOUNDS)
+	dir, err := ioutil.TempDir("", "tracee-make")
+	if err != nil {
+		return err
+	}
+	if debug {
+		fmt.Printf("building bpf object in: %s\n", dir)
+	} else {
+		defer os.RemoveAll(dir)
+	}
+	objFile := filepath.Join(dir, "tracee.bpf.o")
+	err = unpackBPFBundle(dir)
+	if err != nil {
+		return err
+	}
+	clang := locateFile("clang", []string{os.Getenv("CLANG")})
+	if clang == "" {
+		return fmt.Errorf("missing compilation dependency: clang")
+	}
+	llc := locateFile("llc", []string{os.Getenv("LLC")})
+	if clang == "" {
+		return fmt.Errorf("missing compilation dependency: llc")
+	}
+	llvmstrip := locateFile("llvm-strip", []string{os.Getenv("LLVM_STRIP")})
+	if clang == "" {
+		return fmt.Errorf("missing compilation dependency: llvm-strip")
+	}
+	kernelSource := locateFile("", []string{os.Getenv("KERN_SRC"), fmt.Sprintf("/lib/modules/%s/build", tracee.UnameRelease())})
+	if kernelSource == "" {
+		return fmt.Errorf("missing compilation dependency: kernelSource")
+	}
+	linuxArch := os.Getenv("ARCH")
+	if linuxArch == "" {
+		linuxArch = strings.Replace(runtime.GOARCH, "amd64", "x86", 1)
+	}
+
+	// from the Makefile:
+	// $(CLANG) -S \
+	// 	-D__BPF_TRACING__ \
```

Comment:
I think this comment is redundant as it is clear from the code

Reply:
Since this code section is basically a port of 3 commands from the makefile, I noted the original commands in context with the code that ports them. I want to make sure that the build from make and the build from go is *exactly* the same, and that if someone touches this code it's clear that they either need to change the makefile as well or they should reconsider their change because this is how it compiles from the makefile.  I also think we should update this comment if we change the makefile in the future.